### ISSUE TITLE
[Sourcing-2][Accessibility: ACC-264.1][WIP] Table: Row details not associated to header

### ISSUE THREAD

**Issue by JyotiKumari1507** [SAP/fundamental-ngx#11266]
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
Component having issue: https://sap.github.io/fundamental-ngx/#/platform/table/basic
When the user access a single row of the table , it is observed that the row values are not associated to its header and only the cell values are mentioned. The row values of the table should always be associated to header to get the complete information.

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
All latest version

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Using Screen reader can check by implementing similar case. Just use table and access a single row of the table , it will be observe that the row values are not associated to its header and only the cell values are mentioned by screen reader

#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations. 
Yes we have got this bug from Accessibility expert only.

#### Did you check the documentation and the API?
Yes

**Comment by nikolay-kolarov** [comment]
Hi @JyotiKumari1507,

When I tried to read the table, I get announcement for the row count.
What exactly do you mean with row details? Could you please give more info?

Thanks,
Nikolay

**Comment by JyotiKumari1507** [comment]
@droshev Please refer attached screen shot of screen reader. Its reading table cell value after table name. it is observed that the row values are not associated to its header. As per accessibility team row values of the table should always be associated to header to get the complete information

<img width="553" alt="image" src="https://github.com/SAP/fundamental-ngx/assets/157268666/c5c3f335-037a-485d-a840-4113b26a68e5">

**Comment by nikolay-kolarov** [comment]
Hi @JyotiKumari1507,

It is strange when you give another example now. Could you please the initial sample (https://sap.github.io/fundamental-ngx/#/platform/table/basic) and give concrete steps how to navigate in the table and what you expect?

Thanks,
Nikolay

**Comment by JyotiKumari1507** [comment]
@droshev  As per accessibility team, it is observed that the row values of tables are not associated to its header. and this is issue. If required i can ask recording from accessibility team

**Comment by nikolay-kolarov** [comment]
Hello @JyotiKumari1507,

It is hard to work over pictures or recordings.
We need info about the exactly used components (a live sample) and steps how to reproduce.

On the picture that you sent:
- I do not understand if this is a table or list.
- I do not understand how the user navigated to get this output.
- I do not understand if the "Name" cell is focusable and has been focused during the navigation.
- In the screen reader announcement I see more information that I see on the visible UI, so I cannot understand by the picture why this is the case.

If I test the link that you gave (https://sap.github.io/fundamental-ngx/#/platform/table/basic) I am able to read the column header and they are announced when navigated between columns.

Best Regards,
Nikolay

**Comment by JyotiKumari1507** [comment]
@nikolay-kolarov its component https://sap.github.io/fundamental-ngx/#/platform/table/basic) only.
You can see accessibility team screen reader report in the screenshot which i attached earlier. Screen reader able to read table each sell value , but not header.
html used code:

<fdp-table
  #table
  id="appTable"
  [dataSource]="source"
  [noVerticalBorders]="true"
  [noBorderX]="true"
  [noBorderY]="true"
  [rowsActivable]="true"
  [rowNavigatable]="true"
  emptyTableMessage="No Applications found"
  (rowActivate)="onRowActivate($event)"
  emptyTableMessage="No Applications found."
  [initialSortBy]="[
    { field: 'updateAtLabel', direction: sortDirectionEnum.DESC }
  ]"
  bodyHeight="{{
    source.dataProvider.totalItems >= 16 && isCustomerTab
      ? '740px'
      : source.dataProvider.totalItems >= 10 && !isCustomerTab
      ? '484px'
      : 'auto'
  }}"
  fd-scrollbar
  noHorizontalScroll
>
  <fdp-table-toolbar
    title="Applications"
    [shouldOverflow]="true"
  ></fdp-table-toolbar>

  <fdp-column
    name="appName"
    key="appName"
    label="{{ nameLabel }}"
    [sortable]="true"
    [filterable]="true"
  >
    <fdp-table-cell *fdpCellDef="let item">
      <div class="blue-bold">
        {{ item.appName }}
      </div>

      <div>
        {{ item.appId }}
      </div>
    </fdp-table-cell>
  </fdp-column>
  <fdp-column
    name="serviceName"
    key="serviceName"
    label="{{ serviceLabel }}"
    [sortable]="true"
    [filterable]="true"
  >
  </fdp-column>
  <fdp-column
    name="createdBy"
    key="createdBy"
    label="{{ addedByLabel }}"
    [groupable]="true"
    [sortable]="true"
    [filterable]="true"
  >
  </fdp-column>
  <fdp-column
    name="createdAt"
    key="createdAtLabel"
    label="{{ addedOnLabel }}"
    [sortable]="true"
    [groupable]="true"
    [filterable]="true"
    align="end"
  >
    <fdp-table-cell *fdpCellDef="let item; as: source">
      {{ item.createdAtLabel }}
    </fdp-table-cell>
  </fdp-column>

  <fdp-column
    name="updateAt"
    key="updateAtLabel"
    label="{{ changedOnLabel }}"
    [sortable]="true"
    [groupable]="true"
    [filterable]="true"
    align="end"
  >
    <fdp-table-cell *fdpCellDef="let item; as: source">
      {{ item.updateAtLabel }}
    </fdp-table-cell>
  </fdp-column>
</fdp-table>


<fdp-table-view-settings-dialog [table]="table" [allowDisablingSorting]="false">
  <fdp-table-view-settings-filter
    column="appName"
    [type]="filterTypeEnum.MULTI"
    label="{{ nameLabel }}"
    [values]="appNameFilteringValues"
  ></fdp-table-view-settings-filter>
  <fdp-table-view-settings-filter
    column="serviceName"
    [type]="filterTypeEnum.MULTI"
    label="{{ serviceLabel }}"
    [values]="serviceNameFilteringValues"
  ></fdp-table-view-settings-filter>
  <fdp-table-view-settings-filter
    column="createdBy"
    [type]="filterTypeEnum.SINGLE"
    label="{{ addedByLabel }}"
    [values]="createByFilteringValues"
  ></fdp-table-view-settings-filter>
  <fdp-table-view-settings-filter
    column="createdAt"
    [type]="filterTypeEnum.SINGLE"
    label="{{ addedOnLabel }}"
    [values]="createAtFilteringValues"
  >
  </fdp-table-view-settings-filter>
  <fdp-table-view-settings-filter
    column="updateAt"
    [type]="filterTypeEnum.SINGLE"
    label="{{ changedOnLabel }}"
    [values]="updatedAtFilteringValues"
  >
  </fdp-table-view-settings-filter>
</fdp-table-view-settings-dialog>

**Comment by droshev** [comment]
Most of the fields are not filled.

**Comment by JyotiKumari1507** [comment]
@droshev can you pls let me know what input field expecting from us?

**Comment by droshev** [comment]
@JyotiKumari1507 Can you create a stackblitz example, document steps and document what your expectation is?

**Comment by droshev** [comment]
@JyotiKumari1507 Can you create a stackblitz example, steps to reproduce and what the expectation is? Otherwise we will close the ticket.

**Comment by divyamalhotra488** [comment]
Hi @droshev, you can try with the example [here](https://stackblitz.com/edit/prywkj?file=package.json)

**Comment by droshev** [comment]
@divyamalhotra488 can you also answer the other part of my previous post! thank you

**Comment by divyamalhotra488** [comment]
Hi @droshev, open the app with screen reader and try to read the table values.
It is observed that the row values are not associated to its header. As per accessibility team row values of the table should always be associated to header to get the complete information

Please refer

![image](https://github.com/user-attachments/assets/22558522-90a8-4a1b-83d3-a731e75aba37)

**Comment by droshev** [comment]
@divyamalhotra488 can you documents steps to reproduce and what the expectation is based on the stackblitz example you have provided?

**Comment by divyamalhotra488** [comment]
@droshev, open the stackblitz example and try to read the table rows using screen reader.
**Expectation:**
Table cell values should be announced as part of associated column header.

**Comment by droshev** [comment]
@divyamalhotra488 I followed the steps, opened stackblitz and use the keyboard navigation over the table and you can reach and read any cell:
<img width="487" alt="Screenshot 2024-08-06 at 09 58 25" src="https://github.com/user-attachments/assets/39ea5499-4107-4f00-9981-a0520565a938">


@nikolay-kolarov Do you think something else is needed?

**Comment by nikolay-kolarov** [comment]
Hi @droshev,
Hi @divyamalhotra488,

I also checked it and in my understanding this is working fine and the column headers are announced correctly when you enter or change a column.

https://github.com/user-attachments/assets/85eed245-ca65-4753-be36-82404b32b9e7

So I see not issue.

Best Regards,
Nikolay

**Comment by MeghanaVernekar** [comment]
Hello Colleagues,

Retested the issue and issue still exists. Video evidence attached.

https://github.com/user-attachments/assets/d0bdfba3-ceb3-4d70-9970-9975ba972284

Regards,
Meghana.

**Comment by nikolay-kolarov** [comment]
Hello @MeghanaVernekar,

The video that you provide is for a different example and table?
Please provide a respective example, as the last provided stackblitz is different and as you can see on my video, the problem is not reproducible.

@droshev - FYI

Thanks,
Nikolay

**Comment by divyamalhotra488** [comment]
Created new issue: https://github.com/SAP/fundamental-ngx/issues/12493
